lib/samples/set_reference_scale/set_reference_scale_sample.dart

```diff
@@ -0,0 +1,225 @@
+//
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:flutter/material.dart';
+
+import '../../utils/sample_state_support.dart';
+
+class SetReferenceScaleSample extends StatefulWidget {
+  const SetReferenceScaleSample({super.key});
+
+  @override
+  State<SetReferenceScaleSample> createState() => _SetReferenceScaleState();
+}
+
+class _SetReferenceScaleState extends State<SetReferenceScaleSample>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // Create a list of dropdown menu items to load all the reference scales.
+  final _referenceScaleList = <DropdownMenuItem<double>>[];
+  // Create a list of selected feature layers.
+  final _selectedFeatureLayers = <String>[];
+  // Create a list of all feature layers.
+  late List<String> _allFeatureLayers;
+  // Create a variable to store the map.
+  var _map = ArcGISMap();
+  // Create a variable to store the scale.
+  var _scale = 250000.0;
+  // Create a flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // Create a regular expression to format the scale.
+  final reg = RegExp(r'(\d{1,3})(?=(\d{3})+(?!\d))');
+
+  @override
+  void initState() {
+    super.initState();
+    // Add scales to the list.
+    _referenceScaleList.addAll([
```

Comment:
Now that you have a mechanism to do `replaceAllMapped(reg, mathFunc)`, maybe that can be used here, for simplicity and consistency. Something like:
```dart
for (final value in [500000.0, 250000.0, 100000.0, 50000.0]) {
  _referenceScaleList.add(const DropdownMenuItem(
      value: value,
      child: Text('1:${value.toInt().toString().replaceAllMapped(reg, mathFunc)}'),
    ),
  );
}
```
And you could maybe make a helper function that does the whole `'1:${value.toInt().toString().replaceAllMapped(reg, mathFunc)}'` thing. Regexes are notoriously hard to understand so it might be nice to bundle it up into a helper function with a friendly comment. Maybe like `String formatAsScale(double value) { ... }`.

Reply:
yes this would make sense!